docs/user-guide/migrate-to-greptimedb/migrate-from-clickhouse.md

```diff
@@ -0,0 +1,237 @@
+
+**2. Choose Between Tag and Field Columns**
+- Selecting tag columns (TAG): It is recommended to use highly reused and frequently filtered fields—such as host name, service name, region, or instance ID—as tag columns.
+  - Tags are suitable for low-cardinality and well-categorized fields, and can significantly accelerate filtering.
+- Field (data column): Actual observations, metrics, log message fields, etc. are suitable as fields; high-cardinality fields should be avoided for use as primary keys or tags.
+
+**3. Primary Key and Wide Table Recommendations**
+- Primary Key: Combine tag columns. It’s not recommended to include high-cardinality fields such as log IDs, user IDs or UUIDs to avoid primary key bloat, excessive write amplification, and inefficient queries.
+- Wide Table vs. Multiple Tables: For multiple metrics collected at the same observation point (such as on the same host), it’s better to use a wide table, which improves batch write efficiency and compression ratio.
+
+**4. Index Planning**
+- Inverted Index: Build indexes for low-cardinality tag columns to improve filter efficiency.
+- Skipping/Fulltext Index: Use as needed; avoid building unnecessary indexes on high-cardinality or highly variable fields.
+- Read [Data Index](/user-guide/manage-data/data-index.md) for more info.
+
+Example ClickHouse table structure:
+```sql
+CREATE TABLE example (
+ timestamp DateTime,
+ host String,
+ app String,
+ metric String,
+ value Float64
+ ) ENGINE = MergeTree()
+ ORDER BY (timestamp, host, app, metric);
+```
+
+Recommended table structure after migrating to GreptimeDB:
+```sql
+CREATE TABLE example (
+ `timestamp` TIMESTAMP NOT NULL,
+ host STRING,
+ app STRING INVERTED INDEX,
+ metric STRING INVERTED INDEX,
+ `value` DOUBLE,
+ PRIMARY KEY (app, metric),
+ TIME INDEX (`timestamp`)
+ );
+```
+
+> The choice of tags and the granularity of the time index should be carefully planned in light of your business’s data volume and query scenarios. If the host cardinality is not high (e.g., only a few thousand monitored hosts), you can add it to the primary key and create an inverted index.
+
+---
+
+### Migrating Typical Log Tables
+
+> GreptimeDB already provides built-in modeling for otel log ingestion, so please refer to the [official documentation](/user-guide/ingest-data/for-observability/opentelemetry.md#logs).
+
+Common ClickHouse log table structure:
+```sql
+CREATE TABLE logs
+ (
+ timestamp      DateTime,
+ host           String,
+ service        String,
+ log_level      String,
+ log_message    String,
+ trace_id       String,
+ span_id        String,
+ INDEX inv_idx(log_message)  TYPE ngrambf_v1(4, 1024, 1, 0) GRANULARITY 1
+ ) ENGINE = MergeTree
+ ORDER BY (timestamp, host, service);
+```
+
+Recommended GreptimeDB table structure:
+- Time index: `timestamp` (precision set based on logging frequency)
+- Tag columns: `host`, `service` (fields often used in queries/aggregations)
```

Comment:
The tag concept here is confusing, there is no such "tag" in the logs table DDL below. We better use primary key here.

Reply:
We describe the `tag` concept in the schema design guide. The user must read it at first. It doesn't need to repeat it again everywhere.